server/api/v2/congress.py

```diff
@@ -304,21 +307,43 @@ def get_legislators(cur, score_filter="total", include=None, session_num=curr_se
 	aclu_ids = []
 	term_ids = []
 
+	all_sessions = get_sessions()
+	supported_sessions = {}
+	for num in all_sessions:
+		if num >= first_session:
+			supported_sessions[num] = all_sessions[num]
+
 	rs = cur.fetchall()
 	if rs:
 		for row in rs:
 			aclu_id = row[1]
 			office = 'us_representative' if row[4] == 'rep' else 'us_senator'
+
+			start_date = arrow.get(row[2]).format('YYYY-MM-DD')
+			end_date = arrow.get(row[3]).format('YYYY-MM-DD')
+
+			term_started_late = True
+			term_ended_early = True
+
+			for num in supported_sessions:
+				if supported_sessions[num]['start_date'] == start_date:
+					term_started_late = False
+				if supported_sessions[num]['end_date'] == end_date:
+					term_ended_early = False
```

Comment:
I'm seeing some false positives for both `term_ended_early` and `term_started_late`, specifically for people who have term end dates past the end of the session (ex. Marsha Blackburn of TN with end date in 2025) and term start dates before the beginning of the session. I wonder if we should be comparing the dates in a different way, instead of checking for equivalent term/session dates??

Reply:
Ah you're totally right, good catch. The Senate logic is tricky.